org.eclipse.jdt.core/dom/org/eclipse/jdt/core/dom/TypePattern.java

```diff
@@ -247,12 +248,12 @@ public SingleVariableDeclaration getPatternVariable() {
 	 * Returns the pattern variable of Types Pattern.
 	 *
 	 * @return the pattern variable
-	 * @exception UnsupportedOperationException if this operation is used other than JLS19
+	 * @exception UnsupportedOperationException if this operation is used other than JLS22
 	 * @exception UnsupportedOperationException if this expression is used with previewEnabled flag as false
 	 * @since 3.39
 	 */
 	public VariableDeclaration getPatternVariable2() {
-		supportedOnlyIn20();
+		supportedOnlyIn22();
```

Comment:
Requesting for a cleanup here: Please change this name supportedOnlyIn22() to unSupportedBelow22()

Reply:
I believe the supportedOnlyIn22 method can be handle the logic. Please see the method

```
final void supportedOnlyIn22() {
		if (this.ast.apiLevel < AST.JLS22_INTERNAL) {
			throw new UnsupportedOperationException("Operation only supported in JLS22 AST"); //$NON-NLS-1$
		}
	}
```

Therefore should I create a new method called unSupportedBelow22 ?

Reply:
you can rename the current method if it's used only here

Reply:
Yes, renaming seems appropriate in this case. However, there are already similar methods like supportedOnlyIn22,  supportedOnlyIn20. etc .. Should I still proceed with renaming the method, or would it be better to keep the existing naming convention for consistency?

Reply:
ok. let's not do it for this fix. Please create a new cleanup PR that does this